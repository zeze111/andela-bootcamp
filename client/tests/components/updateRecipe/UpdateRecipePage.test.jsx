import React from 'react';

import connectedUpdateRecipePage, {UpdateRecipePage} from '../../../components/updateRecipe/UpdateRecipePage';
import store from '../../../store';

const props = {
  updateRecipe: jest.fn(() => Promise.resolve()),
  getARecipe: jest.fn(() => Promise.resolve()),
  match: {
    params: {
      recipeId: 3
    }
  },
  recipe: {
    id: 3,
    name: 'something'
  }
}

const setup = () => {
  const wrapper = shallow(<UpdateRecipePage {...props} store={store} />);
  return { wrapper }
}

describe('component will mount', () => {
  it('should call get recipes function', () => {
    const spy = sinon.spy(UpdateRecipePage.prototype, 'componentDidMount');
  });
});

describe('Connect Update Recipe component', () => {
  it('should render component successfully', () => {
    const store = mockStore({
        recipeReducer: {
          recipes: []
        }
    })

    const wrapper = shallow(<UpdateRecipePage {...props} store={store} />);
    expect(wrapper.length).toBe(1);
  });
});

describe('Test for Update Recipe page', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
