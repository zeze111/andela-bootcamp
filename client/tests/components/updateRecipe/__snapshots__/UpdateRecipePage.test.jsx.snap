// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Update Recipe page should render component 1`] = `
Object {
  "wrapper": <div
    id="addRecipebody"
  >
    <main
      id="wrapper"
    >
      <div
        className="container full-container"
      >
        <div
          className="row remove-margin-bottom card-width"
        >
          <div
            className="col s12 m10 l8 push-l2 push-m1 form-padding"
          >
            <div
              className="card z-depth-2"
            >
              <span
                className="card-title col s12 m12 l12 text-headers center-align text-color inner"
              >
                 UPDATE RECIPE
              </span>
              <UpdateRecipeForm
                recipe={
                  Object {
                    "id": 3,
                    "name": "something",
                  }
                }
                updateRecipe={[MockFunction]}
              />
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
}
`;
